packages/quantic/force-app/main/default/lwc/quanticRecommendationList/__tests__/quanticRecommendationList.test.js

```diff
@@ -0,0 +1,485 @@
+      );
+      createTestComponent();
+
+      await flushPromises();
+
+      expect(mockEventHandler).toHaveBeenCalledTimes(1);
+      expect(mockEventHandler).toHaveBeenCalledWith({
+        registerTemplates: functionMocks.registerTemplates,
+      });
+    });
+
+    it('should register the fields to include', async () => {
+      const expectedFormattedFieldsToInclude = ['field1', 'field2'];
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionMocks.loadFieldActions).toHaveBeenCalledTimes(1);
+      expect(functionMocks.loadFieldActions).toHaveBeenCalledWith(
+        exampleEngine
+      );
+      expect(functionMocks.dispatch).toHaveBeenCalledTimes(1);
+      expect(functionMocks.dispatch).toHaveBeenCalledWith(
+        functionMocks.registerFieldsToInclude(expectedFormattedFieldsToInclude)
+      );
+    });
+
+    describe('with custom #numberOfRecommendations option', () => {
+      it('should build the buildRecommendationList controller with the custom number of recommendation value', async () => {
+        createTestComponent({
+          ...defaultOptions,
+          numberOfRecommendations: 5,
+        });
+        await flushPromises();
+
+        expect(functionMocks.buildRecommendationList).toHaveBeenCalledTimes(1);
+        expect(functionMocks.buildRecommendationList).toHaveBeenCalledWith(
+          exampleEngine,
+          {
+            options: {
+              id: defaultOptions.recommendation,
+              numberOfRecommendations: 5,
+            },
+          }
+        );
+      });
+    });
+
+    describe('when invalid property values are passed', () => {
+      it('should diplay an error message when recommendationsPerRow is not a number', async () => {
+        const element = createTestComponent({
+          ...defaultOptions,
+          recommendationsPerRow: 'invalid',
+        });
+        await flushPromises();
+
+        const initializationError = element.shadowRoot.querySelector(
+          selectors.initializationError
+        );
+
+        expect(initializationError).not.toBeNull();
+      });
+
+      it('should diplay an error message when recommendationsPerRow is a negative number', async () => {
+        const consoleErrorSpy = jest
+          .spyOn(console, 'error')
+          .mockImplementation(() => {});
+        const expectedConsoleErrorMessage =
+          'The value of the recommendationsPerRow property must be an integer greater than 0.';
+        const element = createTestComponent({
+          ...defaultOptions,
+          recommendationsPerRow: -1,
+        });
+        await flushPromises();
+
+        const initializationError = element.shadowRoot.querySelector(
+          selectors.initializationError
+        );
+
+        expect(initializationError).not.toBeNull();
+        expect(console.error).toHaveBeenCalledTimes(1);
+        expect(console.error).toHaveBeenCalledWith(expectedConsoleErrorMessage);
+        consoleErrorSpy.mockRestore();
```

Comment:
this should be in a AfterEach of the describe describe('when invalid property values are passed', () => {